cirq-core/cirq/protocols/measurement_key_protocol.py

```diff
@@ -81,6 +100,40 @@ def measurement_key(val: Any, default: Any = RaiseTypeErrorIfNotProvided):
     return measurement_key_name(val, default)
 
 
+def measurement_key_obj(val: Any, default: Any = RaiseTypeErrorIfNotProvided):
```

Comment:
Is the plan still to cycle the names such that (in Cirq v1.0) `measurement_key` returns the object? If so, please leave a TODO here to track that rename.

Reply:
I wanted to avoid the confusion caused due to adding so many methods and then cycling them. So I was waiting for the 0.13 release so that I could directly reuse `measurement_key[s]` and not add new methods for returning objects. I added this PR since it might be useful in the classically controlled implementation. I'm not fully sure how blocking this is for that. @daxfohl to comment. 

Also, there was some discussion of doing a 0.13 release soon in the last Cirq sync. If this PR can wait for the release, it would avoid one cycle of adding removing methods and all the clutter due to having 6 different measurement key protocols.

Reply:
No rush on my side unless there is somebody desperately waiting on classical control that I'm not aware of.

Reply:
Cirq v0.13 will likely land in mid-October at the earliest - there's an internal blocker related to #4511.

As long as we're still traking the `measurement_key` update somewhere, a TODO isn't strictly required here.